packages/react-data-grid/src/HeaderCell.js

```diff
@@ -83,42 +87,54 @@ class HeaderCell extends React.Component {
   };
 
   setScrollLeft = (scrollLeft) => {
-    let node = ReactDOM.findDOMNode(this);
-    node.style.webkitTransform = `translate3d(${scrollLeft}px, 0px, 0px)`;
-    node.style.transform = `translate3d(${scrollLeft}px, 0px, 0px)`;
+    const node = ReactDOM.findDOMNode(this);
+    if (node) {
+      node.style.webkitTransform = `translate3d(${scrollLeft}px, 0px, 0px)`;
+      node.style.transform = `translate3d(${scrollLeft}px, 0px, 0px)`;
+    }
   };
 
   removeScroll = () => {
-    let node = ReactDOM.findDOMNode(this);
+    const node = ReactDOM.findDOMNode(this);
     if (node) {
-      let transform = 'none';
+      const transform = 'none';
       node.style.webkitTransform = transform;
       node.style.transform = transform;
     }
   };
 
   render() {
-    let resizeHandle;
-    if (this.props.column.resizable) {
-      resizeHandle = (<ResizeHandle
-      onDrag={this.onDrag}
-      onDragStart={this.onDragStart}
-      onDragEnd={this.onDragEnd}
-      />);
-    }
-    let className = joinClasses({
+    const { column, rowType } = this.props;
+    const resizeHandle = column.resizable && (
+      <ResizeHandle
+        onDrag={this.onDrag}
+        onDragStart={this.onDragStart}
+        onDragEnd={this.onDragEnd}
+      />
+    );
+    const className = classNames({
       'react-grid-HeaderCell': true,
       'react-grid-HeaderCell--resizing': this.state.resizing,
-      'react-grid-HeaderCell--frozen': columnUtils.isFrozen(this.props.column)
-    });
-    className = joinClasses(className, this.props.className, this.props.column.cellClass);
-    let cell = this.getCell();
-    return (
+      'react-grid-HeaderCell--frozen': columnUtils.isFrozen(column)
+    }, this.props.className, column.cellClass);
+    const cell = (
       <div className={className} style={this.getStyle()}>
-        {cell}
+        {this.getCell()}
         {resizeHandle}
       </div>
     );
+
+    if (rowType === HeaderRowType.HEADER && column.draggable) {
```

Comment:
Wrapping `HeaderCell` content. This has a few benefits
1) `setScrollLeft` is not required in `DraggableHeaderCell`. So this works with frozen columns, it was broken before
2) It works with any header cell renderer

In the future I would like to clean up header rendering logic so we can easily compose them

```
DraggableHeaderCell(SortableHeaderCell(SimpleHeaderCell or column.headerRenderer))
```